### PR TITLE
feat: add example react app to test wallet plugin

### PR DESCRIPTION
**Pull request by krzysu**
based on Dan's https://github.com/web3/create-web3js-dapp

### CODE REVIEW

File: examples/react/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2024 web3js-react-dapp-demo
```
Comment (danforbes) [inline examples/react/LICENSE:3]: Is it important to change this?

File: examples/react/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- [React Docs](https://react.dev/learn)
+
+This project was bootstrapped with
+[Create React App](https://github.com/facebook/create-react-app).
+
+## Getting Started
+
+Install the project dependencies with a dependency manager like npm or Yarn. Use
+`npm start` to start a local development server and navigate to
+http://localhost:3000 to view the dApp.
```
Comment (danforbes) [inline examples/react/README.md:16]: Would be nice to update this.

File: examples/react/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "web3js-react-dapp-demo",
+  "version": "0.1.0",
+  "description": "Demonstration Web3.js + React dApp",
```
Comment (danforbes) [inline examples/react/package.json:4]: I would update the name and description here

File: examples/react/public/index.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web3.js + React demonstration dApp"
```
Comment (danforbes) [inline examples/react/public/index.html:10]: Would be nice to update

File: examples/react/public/index.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>Web3.js + React Demonstration dApp</title>
```
Comment (danforbes) [inline examples/react/public/index.html:26]: Would be nice to update

File: examples/react/public/manifest.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "Web3.js + React dApp",
+  "name": "Web3.js + React Demonstration dApp",
```
Comment (danforbes) [inline examples/react/public/manifest.json:3]: Would be nice to update

File: examples/react/src/App.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    web3Context.web3.eth.getChainId().then(updateChainId);
+    web3Context.web3.eth.net.getId().then(setNetworkId);
+    provider.on("chainChanged", updateProviderIds);
+    return () => provider.removeListener("chainChanged", updateProviderIds);
+  }, [web3Context.currentProvider, web3Context.web3.eth]);
+
+  return (
+    <main>
+      <h1>Web3.js + React Minimal dApp Template</h1>
```
Comment (danforbes) [inline examples/react/src/App.tsx:45]: I need to change this in the template!

**Review by danforbes — APPROVED** [review]
Very nitpicky comments about updating boilerplate. Glad the template can be helpful!

**Comment by danforbes** [comment]
Why doesn't the example dApp actually use the plugin?

**Comment by krzysu** [comment]
@danforbes thanks for the review, I will apply all your comments in a follow up PR https://github.com/web3/web3-wallet-rpc-utils/pull/5 as I'm afraid the changes might get lost as I restructured a lot of files later. The usage of a plugin is coming in later PRs as well.

File: examples/react/LICENSE (hunk 1 above)
Comment (krzysu) [reply examples/react/LICENSE:3]: I removed example app license as I don't think it is relevant here, I kept plugin license in the repo's root

File: examples/react/README.md (hunk 2 above)
Comment (krzysu) [reply examples/react/README.md:16]: what would you like to update here? the example app's purpose is to allow for manual tests of the wallet rpc plugin with different extension wallets

Comment (danforbes) [reply examples/react/README.md:16]: I created this PR https://github.com/web3/web3-wallet-rpc-utils/pull/10

**Pull request merged**